marconi-mamba/test/Spec/Marconi/Mamba/Api/Query/Indexers/Utxo.hs

```diff
@@ -136,3 +158,82 @@ generateUtxoRowJson = do
                 (Just $ C.hashScript script)
         utxoRow = UtxoRow utxo (C.SlotNo 1) blockHeaderHash
     pure $ Aeson.encodePretty utxoRow
+
+-- | Start a servant process and execute the client action
+withUserApp
+  :: IndexerEnv
+  -> (Warp.Port -> IO (JsonRpcResponse String UtxoQueryResult)) -- ^ hoisted JSON-RPC client such that it can run in IO
+  -> IO (JsonRpcResponse String UtxoQueryResult)
+withUserApp env  = Warp.testWithApplication(pure $ marconiApp env)
+
+-- | Create http JSON-RPC client, and execute the JsonRpc functions
+clientEnv
+  :: String                                       -- ^ bech32 ShelleyEra address to fetch Utxo's for
+  -> Warp.Port                                    -- ^ JSON-RPC port the server will respond to
+  -> IO (JsonRpcResponse String UtxoQueryResult)  -- ^ response receive from the JSON-RPC server
+clientEnv address port = do
+  manager' <- newManager defaultManagerSettings
+  let env = mkClientEnv manager' (baseUrl port)
+  let (rpcEcho :<|> rpcTargets :<|> rpcUtxos) = getClients env
+  -- test will fail if these calls/routes fail.
+  _ <- rpcEcho "client calling "  -- we just want to know if the route is alive
+  _ <- rpcTargets ""              -- we just want to know if the route is alive
+  rpcUtxos address
+
+-- | Test round trip Utxos thruough JSON-RPC http server
+-- We compare a represenation of the generated UtxoEvents
+-- with those fetched from the JSON-RPC  server. The purpose of this is:
+--   + RPC server routes the request to the correct handler
+--   + Events are serialized/deserialized thru the RPC layer and it various wrappers correctly
+
+jsonRpcTest :: Property
+jsonRpcTest =  property $ do
+  event <- forAll genEvent
```

Comment:
Should use `genUtxoEvents` instead.

Reply:
That is what I used originally,  However, genUtxoEvents generates both Shelley and Byron era addresses and added complexity to the test criteria as currently we only[ query for Shelley addresses](https://github.com/input-output-hk/plutus-apps/blob/main/marconi-mamba/src/Marconi/Mamba/Api/Query/Indexers/Utxo.hs#L60)

Reply:
@koslambrou , forgot to tag you on my response above. Do you still want me to change the generator to getUtxoEvents?

Reply:
Mmmm... Yes, I think you need to use `genUtxoEvents`. You can extend `genUtxoEvents` to only generate Shelley addresses. Probably adding a `Gen Address` as a parameter?

Reply:
Or maybe better, just filter the UtxoEvents of `genUtxoEvents` that has Byron addresses